proposals/0000-add-support-for-css-color-function.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Add support for CSS color() function
+author:
+  - Ryan Linton
+  - Yulian Glukhenko
+date: 2023-11-20
+---
+
+# RFC0000: Add support for CSS color() function
+
+## Summary
+
+React Native does [not currently support](https://github.com/facebook/react-native/issues/41517) wide color gamut color spaces (i.e. display-p3). This proposal discusses adding support for the CSS color() function enabling support for additional wide-gamut color spaces.
+
+## Basic example
+
+```js
+StyleSheet.create({
+  backgroundColor: "color(display-p3 1 0.5 0)",
+  color: "color(display-p3 1 0.5 0 / .5)",
+});
+```
+
+## Motivation
+
+Since most new devices support wider gamut color spaces, React Native should support them as well. The Display P3 color space has had native support since Android 8.0 and iOS 9.3. The color() function was introduced with [CSS Color Module Level 4](https://drafts.csswg.org/css-color/#color-function), much of which is already implemented in React Native. Also Flutter [recently added support](https://github.com/flutter/flutter/issues/55092) for Display P3 on iOS with plans to follow up with support for Android and then the framework itself.
+
+## Detailed design
+
+### JS Changes
+
+1. Parse the color() function in [normalizeColor](https://github.com/facebook/react-native/blob/63213712125795ac082597dad2716258b90cdcd5/packages/normalize-color/index.js#L235)
+
+```js
+const COLOR_SPACE = /display-p3|srgb/
```

Comment:
does the `color` function force you to specify the color space or assume some default?

Reply:
It forces you to specify a [predefined color space](https://www.w3.org/TR/css-color-4/#color-function). When I try `background: color(0,0,1)` in Chrome I get "invalid property value".